src/server/modes/charts/plugins/datalens/utils/color-helpers.ts

```diff
@@ -271,6 +272,32 @@ function mapAndColorizeGraphsByMeasure(
     }
 }
 
+function mapAndColorizePieByMeasure(
+    points: (PiePoint & ExtendedSeriesLineOptions)[],
+    colorsConfig: ChartColorsConfig,
+) {
+    const colorValues = points.map((point) => Number(point.colorValue) as ColorValue);
+
+    const gradientThresholdValues = getThresholdValues(colorsConfig, colorValues);
+    const gradientColors = getColorsByMeasureField({
+        values: colorValues,
+        colorsConfig,
+        gradientThresholdValues,
+    });
+
+    if (gradientThresholdValues.range !== 0) {
```

Comment:
The gradient should be applied even if the range is equal to 0 (with a single measure value)